### DISCUSSION TITLE
How to close the Popover programatically?

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [radix-ui/primitives#2054] [Help]
Hi,

I have a bunch of buttons in a popover and whenever anyone of them is clicked, I want to close the popover they're in.

How do I do that in Radix?

Thanks

**Comment by joaom00** [comment]
You can do it controlling the state

```tsx
const [open, setOpen] = React.useState(false)

<Popover.Root open={open} onOpenChange={setOpen}>
  ...
</Popover.Root>
```

**Reply by andy-hook** [reply]
Alternatively, you can wrap your buttons in `<Popover.Close />` and pass `asChild`

https://www.radix-ui.com/docs/primitives/components/popover#close
